### PR TITLE
Adding tryFindViewElement to testing docs

### LINKED ISSUE #497: Add the tryFindViewElement function to the documentation
Currently the documentation on unit testing only mention the Viewer APIs.
It should also mention how to use `tryFindViewElement` to write unit tests for views

### PR DESCRIPTION
**Pull request by SergejDK**
fix #497

Hopefully I got everything right here....

### CODE REVIEW

File: docs/Fabulous.XamarinForms/testing.md [hunk 1]
```diff
@@ -151,6 +152,22 @@ let ``View should generate a valid interface``() =
     stepSizeLabel.Text |> should equal "Step size: 4"
 ```
 
+We can use `tryFindViewElement` to check if a ViewElement is available. We can use it as follows with the view which we declared before:
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]: Maybe we should be a little bit clearer.

`tryFindViewElement` is not really here to check if a ViewElement exists, but more for quick access to a ViewElement by using its AutomationId.
It's an alternative to the verbose Viewer API:

```fsharp
let contentPage = ContentPageViewer(actualView)
let stackLayout = StackLayoutViewer(contentPage.Content)
let countLabel = LabelViewer(stackLayout.Children.[0])
let timerSwitch = SwitchViewer(StackLayoutViewer(stackLayout.Children.[3]).Children.[1])
let stepSlider = SliderViewer(stackLayout.Children.[4])
let stepSizeLabel = LabelViewer(stackLayout.Children.[5])
```
vs
```fsharp
let stepSizeLabelOpt = tryFindViewElement "stepSizeLabel" actualView
```

So, maybe add 2 sublevels (`####`) presenting (1) how to use the Viewer API's way of accessing things and (2) `tryFindViewElement`?

Checking the existence of a control is only an added nice thing to `tryFindViewElement` that we can mention. :)

Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]: Also we should mention the `findViewElement` helper

**Comment by SergejDK** [comment]
tried to make the difference clearer.

File: docs/Fabulous.XamarinForms/testing.md [hunk 2]
```diff
@@ -151,6 +156,35 @@ let ``View should generate a valid interface``() =
     stepSizeLabel.Text |> should equal "Step size: 4"
 ```
 
+#### FindViewElement / TryFindViewElement
+With `findViewElement` and `tryFindViewElement` you don't need to know where exactly the child is positioned. You have to set `automationId` on the ViewElements which will be used by those functions to find the element in the tree. 
+This approach is the recommended way for testing and to get the ViewElements in a View.
+
+##### findViewElement
+```fsharp
+let actualView = App.view  model  ignore
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]:
```suggestion
[<Test>]
let ``View should generate a label showing the count number of the model``() =
    let model = { Count = 5; Step = 4; TimerOn = true }
    let actualView = App.view  model  ignore
    
    let countLabel = findViewElement "CountLabel" actualView |> LabelViewer
    
    countLabel.Text |> should equal "5"
```

File: docs/Fabulous.XamarinForms/testing.md [hunk 3]
```diff
@@ -108,6 +109,7 @@ Let's take this code for example:
 let view (model: Model) dispatch =  
     View.ContentPage(
         content=View.StackLayout(
+            automationId="stackLayoutId" // used for tryFindViewElement and findViewElement
             children=[ 
                 View.Label(text=sprintf "%d" model.Count)
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]:
```suggestion
                View.Label(automationId="CountLabel", text=sprintf "%d" model.Count)
```

File: docs/Fabulous.XamarinForms/testing.md [hunk 4]
```diff
@@ -132,6 +134,9 @@ From there, we create the Viewers to help us read the properties of the controls
 
 And finally, we assert that the properties have the expected values.
 
+#### Viewer API
+The following approach uses the Viewer API. This is a way but with this you have to know exactly at which position the child you need is. 
+
 ```fsharp
 [<Test>]
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]:
```suggestion
[<Test>]
let ``View should generate a label showing the count number of the model``() =
    let model = { Count = 5; Step = 4; TimerOn = true }
    let actualView = App.view  model  ignore
    
    let contentPage = ContentPageViewer(actualView)
    let stackLayout = StackLayoutViewer(contentPage.Content)
    let countLabel = LabelViewer(stackLayout.Children.[0])
    
    countLabel.Text |> should equal "5"
```

File: docs/Fabulous.XamarinForms/testing.md [hunk 5]
```diff
@@ -151,6 +156,35 @@ let ``View should generate a valid interface``() =
...
+let stacklayout = actualView 
+		|> findViewElement "stackLayoutId" 
+		|> StackLayoutViewer
+```
+
+##### tryFindViewElement
+`tryFindViewElement` delivers a quickaccess to a ViewElement as findViewElement but here you get an Option Type. With this you can also check for the existence of a ViewElement. 
+
+```fsharp
+ [<Test>]
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]:
```suggestion
[<Test>]
let ``When user is authenticated, View should not include a connection button``() =
    let model = { Count = 5; Step = 4; TimerOn = true }
    let actualView = App.view  model  ignore
    
    tryFindViewElement "ConnectionButton" actualView |> should equal None
```

File: docs/Fabulous.XamarinForms/testing.md [hunk 6]
```diff
@@ -108,6 +109,7 @@ Let's take this code for example:
 let view (model: Model) dispatch =  
     View.ContentPage(
         content=View.StackLayout(
+            automationId="stackLayoutId" // used for tryFindViewElement and findViewElement
```
Comment (TimLariviere) [inline docs/Fabulous.XamarinForms/testing.md]: If my suggestion is applied, this line will need to be reverted

**Pull request merged**